core/src/main/java/org/jdbi/v3/core/result/ResultIterable.java

```diff
@@ -81,6 +82,40 @@
     @Override
     ResultIterator<T> iterator();
 
+    /**
+     * Returns a {@code ResultIterable<U>} derived from this {@code ResultIterable<T>}, by
+     * transforming elements using the given mapper function.
+     *
+     * @param mapper function to apply to elements of this ResultIterable
+     * @param <U>    Element type of the returned ResultIterable
+     * @return the new ResultIterable
+     */
+    default <U> ResultIterable<U> map(Function<? super T, ? extends U> mapper) {
+        return () -> new ResultIterator<U>() {
+            ResultIterator<T> delegate = iterator();
```

Comment:
should be `private final`, no? The return signature will prevent it from being accessed, but still, if it was a class in its own right we wouldn't allow it.

Reply:
Yep. I'll push another change

Reply:
Addressed in #1530